src/Extracting/DatabaseTransactionHelpers.php

```diff
@@ -30,9 +49,60 @@ private function endDbTransaction()
 
         foreach ($connections as $conn) {
             try {
-                app('db')->connection($conn)->rollBack();
+                tap(
+                    app('db')->connection($conn),
+                    function ($driver) use ($conn) {
+                        if (self::driverSupportsTransactions($driver)) {
+                            return $driver->rollBack();
+                        }
+
+                        c::warn("Database driver [{$conn}] does not support transactions!");
```

Comment:
Terminology. We need to be consistent. Is it the connection or the driver that does not support transactions?🙂

Reply:
Hmmm... Honestly, It could be both, from the little reading around the subject I have done, some connections/databases don't support transactions and, library dependant, some drivers don't support transactions. Seen as an in-process transaction support **could** be developed for those drivers where the connections don't support transactions, I felt it prudent to lay the "blame" at the driver level rather than at the connection.

In the case of `Jenssegers`. MongoDB supports transactions, however Jenssegers has yet to implement this feature. It may also be the case that a user is using an older version of a library when the latest version does indeed support transactions. If that where the case, but the database itself was out-of-date and didn't support transactions, then surely that would be for the driver to handle?

Thoughts?

Reply:
Re-reading this, I think my logic is a bit squirrelly. Basically, I think there are a few things we need to consider:

1. Database supports transactions & driver does not (Most likely)
1. Database and driver do not support transactions (Not likely but possible)
1. Database does not support transactions but driver does (doesn't really affect us)

In terms of our messaging and terminology we have to decide where we lay the fault. As we connect to the database using a driver/adaptor/connector, I think it is prudent to place the fault at that level. 

Relating this to `Illuminate\Database`, the Transactions methods are added to the `Illuminate\Database\Connection` class via a Concern/trait. So the correct terminology in this case would be "Connection" however, if I know nothing about the inner workings of Laravel, would I consider "Connection" as the actual connection to the database server or the software responsible for initiating said connection?

In terms of design patterns, `Illuminate\Database` utilizes the Adaptor pattern for its various "Connection" classes. This would probably be the most correct terminology however, the subclasses of `Illuminate\Database\Connection` aren't stored or referred to as "Adaptors" so would this add confusion? Would the synonym "Driver" be more apt?

Reply:
I have updated the messaging to..
> Database driver for the connection [mongodb] does not support transactions!

This also maps to the message in the `DbTransactionSupportException` with.. 
> Database driver [Jenssegers\Mongodb\Connection] for connection [mongodb] does not support transactions. Changes to your database will be persistent. To allow this, add "mongodb" to the "run_without_database_transactions" config.

Thoughts?

Reply:
Yeah, that's good.

Reply:
Actually, I'm going to lay the blame on the driver. The connection is an arbitrary name the user picks.